.github/workflows/compatibility.yml

```diff
@@ -0,0 +1,124 @@
+name: Cocmpatibility - WC, WP and PHP
+
+on:
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  woocommerce-compatibility:
+    name:    WC compatibility
+    runs-on: ubuntu-16.04
+    strategy:
+      fail-fast:    false
+      max-parallel: 10
+      matrix:
+        woocommerce: ['4.0.0', '4.3.0', '4.4.0', '4.5.0', '4.6.0']
+        wordpress:   ['latest']
+        php:         ['7.4']
+    steps:
+      # clone the repository
+      - uses: actions/checkout@v2
+      # enable dependencies caching
+      - uses: actions/cache@v2
+        with:
+          path: ~/.cache/composer/
+          key:  ${{ runner.os }}-composer-${{ hashFiles('composer.lock') }}
+      # setup PHP, but without debug extensions for reasonable performance
+      - uses: shivammathur/setup-php@v2
+        with:
+          php-version: ${{ matrix.php }}
+          tools:       composer
+          coverage:    none
+      # install dependencies and run tests
+      - run: composer self-update 2.0.6 && composer install --no-progress
+      - run: sudo systemctl start mysql.service
+      - run: WCPAY_DIR="$GITHUB_WORKSPACE" bash ./bin/install-wp-tests.sh woocommerce_test root root localhost ${{ matrix.wordpress }} ${{ matrix.woocommerce }} false
+      - run: bash ./bin/phpunit.sh
+
+  compatibility-oldest:
+    name:    Environment - oldest
+    runs-on: ubuntu-16.04
+    strategy:
+      matrix:
+        woocommerce: ['4.0.0']
+        wordpress:   ['5.3']
+        php:         ['7.0']
+    steps:
+      # clone the repository
+      - uses: actions/checkout@v2
+      # enable dependencies caching
+      - uses: actions/cache@v2
+        with:
+          path: ~/.cache/composer/
+          key:  ${{ runner.os }}-composer-${{ hashFiles('composer.lock') }}
+      # setup PHP, but without debug extensions for reasonable performance
+      - uses: shivammathur/setup-php@v2
+        with:
+          php-version: ${{ matrix.php }}
+          tools:       composer
+          coverage:    none
+      # install dependencies and run tests
+      - run: composer self-update 2.0.6 && composer install --no-progress
+      - run: sudo systemctl start mysql.service
+      - run: WCPAY_DIR="$GITHUB_WORKSPACE" bash ./bin/install-wp-tests.sh woocommerce_test root root localhost ${{ matrix.wordpress }} ${{ matrix.woocommerce }} false
+      - run: wget https://phar.phpunit.de/phpunit-6.5.14.phar && mv phpunit-6.5.14.phar phpunit.phar
+      - run: bash ./bin/phpunit.sh
```

Comment:
Those steps seem to be repeated many times. Is there a chance to reuse them?

Reply:
Yes, with a bash script. I'll implement it!